backend/lib/endpoints/users.js

```diff
@@ -16,6 +17,205 @@ async function routes(app) {
   const User = app.mongo.model("IndividualUser");
   const Post = app.mongo.model("Post");
 
+  const USERS_PAGE_SIZE = 10;
+
+  app.get(
+    "/",
+    {
+      preValidation: [app.authenticateOptional],
+      schema: getUsersSchema,
+    },
+    async (req) => {
+      const { query, userId } = req;
+      const {
+        ignoreUserLocation,
+        filter,
+        keywords,
+        limit,
+        objective,
+        skip,
+        includeMeta,
+      } = query;
+      const queryFilters = filter ? JSON.parse(decodeURIComponent(filter)) : {};
+      let user;
+      let userErr;
+      if (userId) {
+        [userErr, user] = await app.to(User.findById(userId));
+        if (userErr) {
+          req.log.error(userErr, "Failed retrieving user");
+          throw app.httpErrors.internalServerError();
+        } else if (user === null) {
+          req.log.error(userErr, "User does not exist");
+          throw app.httpErrors.forbidden();
+        }
+      }
+
+      /* eslint-disable sort-keys */
+      const filters = [{ type: "Individual" }];
+
+      // prefer location from query filters, then user if authenticated
+      let location;
+      if (queryFilters.location) {
+        location = queryFilters.location;
+      } else if (user && !ignoreUserLocation) {
+        location = user.location;
+      }
+
+      if (queryFilters.location) filters.push({ "hide.address": false });
+
+      const objectives = {
+        request: {
+          $or: [{ "needs.medicalHelp": true }, { "needs.otherHelp": true }],
+        },
+        offer: {
+          $or: [
+            { "objectives.donate": true },
+            { "objectives.shareInformation": true },
+            { "objectives.volunteer": true },
+          ],
+        },
+      };
+
+      if (objective) filters.push(objectives[objective]);
+
+      // if location is defined, use simple regex text query, in order to use $geoNear
+      if (location && keywords) {
```

Comment:
I think we can convert this regex to a util. I saw it very similarly in 3 different places in the backend.

Reply:
and at least once in the FE.

Reply:
Agreed @idjevm for the backend, they look the same so let's make them a function in `backend/lib/utils.js`.

I see the similar front end functions as well. We can move to util on frontend side too. The `Highlight` fn in:

client/src/components/Feed/Post.js
client/src/components/Feed/User.js

Still haven't gotten around to sharing constants/functions across backend & frontend so let's leave them separate for now.

Reply:
moved both on BE and FE to `backend/lib/utils.js` and `components/Feed/utils.js`